types/index.d.ts

```diff
@@ -3,6 +3,6 @@ import _Vue, { Component, AsyncComponent } from 'vue'
 declare function wrap(
   Vue: typeof _Vue,
   Component: Component | AsyncComponent
-): HTMLElement
+): CustomElementConstructor
```

Comment:
```suggestion
): typeof HTMLElement
```

This will fix the return typing issue, and also is more accurate. Here is why:
`wrap()` function [returns this](https://github.com/vuejs/vue-web-component-wrapper/blob/f4c06998ac20585cba7e9b0a5836902e2aa2a50a/dist/vue-wc-wrapper.js#L167), which is indeed a type of `HTMLElement`. This type declaration will indeed fix the lack of congruency.